### PR TITLE
Fix bug in CTA amounts under certain langs

### LINKED ISSUE #4332: Suggested donation banner populates incorrect amount in some other languages
From [FD](https://gratipay.freshdesk.com/helpdesk/tickets/6792):

> I've donated to a project and then when your portal offered my to donate 0,40 to Gratipay I've clicked the button, but then my account shows u$s 4 dollar donation to Gratipay
May be it's a bug or I didn't undestand it. The BUTTON clearly displayed the number 0,40. Then I clicked it.

This works correctly for me. I think this may be a language issue around `,` vs `.` for currency? Screenshot submitted by FD user shows `,` is used.

English:
image

I am not sure how to change my language so I can test this myself.

## notify
- [ ] https://gratipay.freshdesk.com/helpdesk/tickets/6792

### PR DESCRIPTION
**Pull request by whit537**
Closes #4332.

### CODE REVIEW

**Comment by whit537** [comment]
Blech, hitting https://github.com/gratipay/gratipay.com/issues/4322 (I usually use FF for lang manipulations) ...

**Comment by whit537** [comment]
Okay, I've got a test that varies the lang, trying to repro the original bug ...

**Comment by whit537** [comment]
Turns out it's [hard](http://stackoverflow.com/questions/17858663/custom-headers-in-phantomjs-selenium-webdriver) to vary the language in the ttw tests because of limitations in the drivers underlying selenium (which underlies splinter).

**Comment by whit537** [comment]
Here's a [project](https://github.com/glenfant/splinterext.proxyheader) for twiddling headers in a proxy, but it's [not](https://pypi.python.org/pypi/splinterext.proxyheader) in PyPI(?).

**Comment by whit537** [comment]
Alright, the problem is that we [set](https://github.com/gratipay/gratipay.com/blob/2049/templates/support-gratipay.html#L20-L21) `data-amount` w/o regard for locale, and then [send](https://github.com/gratipay/gratipay.com/blob/2049/js/gratipay/payments.js#L38) that to the server and [pass](https://github.com/gratipay/gratipay.com/blob/2049/www/%25team/payment-instruction.json.spt#L31) it into `parse_decimal`, which [_does_](https://github.com/gratipay/gratipay.com/blob/2049/gratipay/utils/i18n.py#L222) vary with locale.

**Comment by whit537** [comment]
So we can either set a locale-aware `data-amount`, or parse_decimal naively.

**Comment by whit537** [comment]
Ready for review, @mattbk @nobodxbodon @JessaWitzel et al.

**Comment by mattbk** [comment]
Travis looks stuck?

**Comment by mattbk** [comment]
This button is gone from /about/pricing:
image

on localhost:
image

**Comment by whit537** [comment]
Ish. The merge commit build was hung (though further into the ttw tests than I've seen before). The other failed on pyflakes. Redoing the commit ...

**Comment by whit537** [comment]
Green!

**Comment by mattbk** [comment]
When using another language, it now gives me the "You have canceled your payment to Gratipay" message and doesn't add a donation.

**Comment by mattbk** [comment]
Although that doesn't make sense, because that part of the code wasn't touched. I think my database is not set up exactly right for this yet.

**Comment by mattbk** [comment]
I'm not clear on what is called when the buttons are pushed. Where is the logic for that?

**Comment by mattbk** [comment]
0a66db5 is what makes it work for me.

**Comment by whit537** [comment]
Hmmm ...

**Comment by mattbk** [comment]
I know it was supposed to work, but didn't for me until I threw out that js function. Will try in another environment.

**Comment by mattbk** [comment]
Works on my Mac too. :man_shrugging:

**Comment by mattbk** [comment]
@nobodxbodon, do you have time to look at this?

File: templates/support-gratipay.html [hunk 1]
```diff
@@ -17,8 +17,12 @@
             {% endif %}
         </td>
         <td class="knobs" rowspan="2">
-            <button class="low" data-amount="{{ cta_low }}">{{ format_currency(cta_low, 'USD', trailing_zeroes=False) }}</button><button
-                class="high" data-amount="{{ cta_high }}">{{ format_currency(cta_high, 'USD', trailing_zeroes=False) }}</button>
+            {% set dlo = format_decimal(cta_low) %}
+            {% set dhi = format_decimal(cta_high) %}
+            {% set clo = format_currency(cta_low, 'USD', trailing_zeroes=False) %}
+            {% set chi = format_currency(cta_high, 'USD', trailing_zeroes=False) %}
+            <button class="low" data-amount="{{ dlo }}">{{ clo }}</button><button
```
Comment (nobodxbodon) [inline templates/support-gratipay.html]: nit: move <button at the end to the next line

Comment (mattbk) [inline templates/support-gratipay.html]: That moves the button itself to a new line, unfortunately.

Comment (nobodxbodon) [inline templates/support-gratipay.html]: Please nvm then. Sorry about that.

**Comment by whit537** [comment]
Rebased, was e1a1b245cf898b1e69d9485dacf80c3eee02f22e.

**Comment by whit537** [comment]
Let me see if I can repro ...

**Comment by whit537** [comment]
There it is. I think I only tested the sidebar giving widget, not the global cta.

!m @mattbk

**Pull request merged**